src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8000,33 +8014,59 @@ private CommonForEachStatementSyntax ParseForEachStatement()
             }
 
             var openParen = this.EatToken(SyntaxKind.OpenParenToken);
+            var variable = ParseExpressionOrDeclaration(ParseTypeMode.Normal, feature: MessageID.IDS_FeatureTuples, permitTupleDesignation: true);
+            var @in = this.EatToken(SyntaxKind.InKeyword, ErrorCode.ERR_InExpected);
+            if (!IsValidForeachVariable(variable))
+            {
+                @in = this.AddError(@in, ErrorCode.ERR_BadForeachDecl);
```

Comment:
I believe that the error is at the right location, but I'm open to an investigation about moving it.

---
In reply to: [90168970](https://github.com/dotnet/roslyn/pull/15548#discussion_r90168970) [](ancestors = 90168970)